pypeit/core/wavecal/autoid.py

```diff
@@ -854,8 +854,46 @@ def reidentify(spec, spec_arxiv_in, wave_soln_arxiv_in, line_list, nreid_min, de
     return detections, spec_cont_sub, patt_dict_slit
 
 
+def measure_fwhm(spec, binspectral):
+    """
+    Measure the arc lines FWHM, i.e, approximate spectral resolution
+
+    Args:
+        spec (`numpy.ndarray`_):
+            Arc spectrum from a single slit
+        binspectral (:obj:`int`):
+          Binning of the input arc in the spectral dimension
+
+    Returns:
+        :obj:`float`: Measured FWHM in pixels
+    """
+
+
+    # Determine the lines FWHM, i.e, approximate spectral resolution
+    #  This may only be recorded and not used by the algorithms
+    _, _, _, wdth, _, best, _, nsig = arc.detect_lines(spec, sigdetect=10., fwhm=5.)
+    # 1sigma Gaussian widths of the line detections
+    wdth = wdth[best]
+    # significance of each line detected
+    nsig = nsig[best]
+    # Nsigma (significance) threshold. We use only lines that have the highest significance
+    # We start with nsig_thrshd of 500 and iteratively reduce it if there are not more than 6 lines
+    nsig_thrshd = 500.
+    measured_fwhm = None
+    while nsig_thrshd > 10.:
+        if wdth[nsig > nsig_thrshd].size > 6:
+            # compute average `wdth`
+            mean, med, _ = stats.sigma_clipped_stats(wdth[nsig > nsig_thrshd], sigma_lower=2.0, sigma_upper=2.0)
+            # FWHM in pixels
+            measured_fwhm = np.round((med * 2.35482) / binspectral,1)
```

Comment:
Following on @profxj's comments, I'm unsure about this conversion, and we should be clear on this in the docs.  If the input spectrum is binned and `binspectral` is the binning of that input spectrum, is this converting the FWHM from the number of binned pixels to the number of unbinned pixels?  If so, shouldn't we be multiplying instead of dividing?

Reply:
@kbwestfall @profxj  thank you for bringing up this confusion about the `fwhm`. Now I am confused too.  I think got confused by the fact that in the doc it says that `['calibrations']['wavelengths']['fwhm']` should be in unbinned pixels, but looking at the code (at least `full_template`) it is not true. In `full_template` the input spectrum is used as it is (binned) and the fwhm is used as it is (which should apparently be binned). I'll check the other methods to be sure, and if I'm correct I'll fix this and the docs.

Reply:
I have checked and it seems fwhm is always used in binned pixels of the input arc frames. I corrected `measured_fwhm` and the docs.